### ISSUE TITLE
iconSize of PointAnnotation results in different sizes depending on device

### ISSUE THREAD

**Issue by lukaskurz** [mapbox/mapbox-maps-flutter#267]
## Expected Behaviour

When drawing markers using the PointAnnotation API, i would expect that annotations drawn on different devices would have the same size.

## Actual Behaviour

It seems that some other factor or variable is affecting the final rendered image size of the icon used with point annotation.

This is the code i use to configure my point annotations
```dart
  Future<void> drawMarkers(List<MarkerWithStyling> markers) async {
    final pointAnnotationManager = await _pointAnnotationManager.future;

    final pointAnnotationOptions = await Future.wait(markers.map((m) => _convertMarkerToPointAnnotation(m)));

    // clear previous polygons and polylines
    await pointAnnotationManager.deleteAll();

    // draw new polygons and polylines
    final points = await pointAnnotationManager.createMulti(pointAnnotationOptions);

...

Future<PointAnnotationOptions> _convertMarkerToPointAnnotation(MarkerWithStyling marker) async {
    final ByteData bytes = await rootBundle.load(marker.iconUrl);
    final Uint8List list = bytes.buffer.asUint8List();

    // get image size
    ui.Codec codec = await ui.instantiateImageCodec(bytes.buffer.asUint8List());
    ui.FrameInfo fi = await codec.getNextFrame();

    final imageSize = max(fi.image.width, fi.image.height);
    const referenceSize = 96;
    final iconSize = referenceSize / imageSize;

    return PointAnnotationOptions(
      geometry: Point(coordinates: marker.position).toJson(),
      iconSize: iconSize,
      iconAnchor: IconAnchor.BOTTOM,
      iconOpacity: 1.0,
      symbolSortKey: 100.0,
      image: list,
    );
  }
```

and these are the results, compared on a iPhone 15 Pro Max Simulator and a Pixel 7 Emulator.

![image](https://github.com/mapbox/mapbox-maps-flutter/assets/22956519/e90540b6-94be-4794-bffd-00fa4e837e5b)

**Comment by lukaskurz** [comment]
from testing it out, the behaviour is similar across emulators, with the experienced difference being almost excactly always 3x of the size on the android screen. I am using version 0.4.5 of your library.

**Comment by Paalpe** [comment]
Experiencing the same thing. Only tested on Ios Physical device. All the pins are huge! it occurred after updating to version 0.4.5.

**Comment by kamami** [comment]
Same here...

**Comment by securexperts** [comment]
Same to Compas Image Size, it's huge too!
mapboxMap.compass.updateSettings(
        CompassSettings(
          visibility: true,
          clickable: true,
          image: uint8List,
          opacity: 0.8,
          marginTop: compasPosition,
          marginRight: 20,
        )
    );

**Comment by baleboy** [comment]
this should be fixed in version 0.5.0, could you confirm?

**Comment by securexperts** [comment]
Yes, I can confirm. Version 0.5.0 is Ok!
